packages/strapi-hook-bookshelf/lib/buildQuery.js

```diff
@@ -0,0 +1,234 @@
+        model,
+        field
+      );
+
+      let fieldKey = `${associationModel.collectionName}.${attributeKey}`;
+
+      if (association && attributeKey === field) {
+        fieldKey = `${associationModel.collectionName}.${associationModel.primaryKey}`;
+      }
+
+      buildWhereClause({
+        qb,
+        field: fieldKey,
+        operator,
+        value,
+      });
+    });
+  }
+
+  if (_.has(filters, 'sort')) {
+    qb.orderBy(
+      filters.sort.map(({ field, order }) => ({
+        column: field,
+        order,
+      }))
+    );
+  }
+
+  if (_.has(filters, 'start')) {
+    qb.offset(filters.start);
+  }
+
+  if (_.has(filters, 'limit') && filters.limit >= 0) {
+    qb.limit(filters.limit);
+  }
+};
+
+/**
+ * Builds a sql where clause
+ * @param {Object} options - Options
+ * @param {Object} options.qb - Bookshelf (knex) query builder
+ * @param {Object} options.model - Bookshelf model
+ * @param {Object} options.field - Filtered field
+ * @param {Object} options.operator - Filter operator (=,in,not eq etc..)
+ * @param {Object} options.value - Filter value
+ */
+const buildWhereClause = ({ qb, field, operator, value }) => {
+  if (Array.isArray(value) && !['in', 'nin'].includes(operator)) {
+    return qb.where(subQb => {
+      for (let val of value) {
+        subQb.orWhere(q => buildWhereClause({ qb: q, field, operator, value: val }));
+      }
+    });
+  }
+
+  switch (operator) {
+    case 'eq':
+      return qb.where(field, value);
+    case 'ne':
+      return qb.where(field, '!=', value);
+    case 'lt':
+      return qb.where(field, '<', value);
+    case 'lte':
+      return qb.where(field, '<=', value);
+    case 'gt':
+      return qb.where(field, '>', value);
+    case 'gte':
+      return qb.where(field, '>=', value);
+    case 'in':
+      return qb.whereIn(field, Array.isArray(value) ? value : [value]);
+    case 'nin':
+      return qb.whereNotIn(field, Array.isArray(value) ? value : [value]);
+    case 'contains': {
+      return qb.whereRaw('LOWER(??) LIKE LOWER(?)', [field, `%${value}%`]);
+    }
+    case 'ncontains':
+      return qb.whereRaw('LOWER(??) NOT LIKE LOWER(?)', [field, `%${value}%`]);
+    case 'containss':
+      return qb.where(field, 'like', `%${value}%`);
+    case 'ncontainss':
+      return qb.whereNot(field, 'like', `%${value}%`);
+
+    default:
+      throw new Error(`Unhandled whereClause : ${field} ${operator} ${value}`);
+  }
+};
+
+/**
+ * Returns a list of model path to populate from a list of where clausers
+ * @param {Object} where - where clause
+ */
+const extractRelationsFromWhere = where => {
+  return where
+    .map(({ field }) => {
+      const parts = field.split('.');
+      return parts.length === 1 ? field : _.initial(parts).join('.');
+    })
+    .sort()
+    .reverse()
+    .reduce((acc, currentValue) => {
+      const alreadyPopulated = _.some(acc, item => _.startsWith(item, currentValue));
+      if (!alreadyPopulated) {
```

Comment:
We didn't talk about the guidelines but I really prefer when there is a new line between constant and condition. It makes the code more readable IMO.